app/pages/notifications/notification-section.jsx

```diff
@@ -143,13 +144,13 @@ export default class NotificationSection extends Component {
     const { notificationsCounter } = this.context;
     const { user } = this.props;
     const { notifications } = this.state;
-    const relatedNotifications = notifications.filter(
-      notification => notification.source_id === readNotification.source_id
-    );
-    relatedNotifications.push(readNotification);
-    const readRequests = relatedNotifications
+    const relatedNotifications = notifications.filter(notification => (
+      notification.subscription_id === readNotification.subscription_id
+        && notification.delivered === false
+    ));
+    const promises = relatedNotifications
       .map(relatedNotification => relatedNotification.update({ delivered: true }).save());
-    Promise.all(readRequests).then(() => notificationsCounter.update(user));
+    Promise.all(promises).then(() => notificationsCounter.update(user));
```

Comment:
Are the promises in `promises` guaranteed to resolve? `Promise.all` will reject if any of them reject and there's no catch block to handle that case. I'm not familiar with the notifications code, but my guess is you want to update the notifications counter after all the API requests complete, even if some failed.

Reply:
The [MDN example of fail-fast behaviour](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all#Promise.all_fail-fast_behaviour) suggests you could add catch blocks to the individual requests too.